modules/assim.sequential/R/load_data_paleon_sda.R

```diff
@@ -0,0 +1,148 @@
+  ## Wish list: eddy covariance
+  
+  library(plyr) #need to load to use .fnc below
+  
+  d <- settings$database$bety[c("dbname", "password", "host", "user")]
+  bety <- src_postgres(host = d$host, user = d$user, password = d$password, dbname = d$dbname)
+  settings$host$name <- "localhost"
+  
+  site <- PEcAn.DB::query.site(settings$run$site$id, bety$con)
+  format_id <- settings$state.data.assimilation$data$format_id
+  
+  input.list <- list()
+  input.id <- list()
+  obvs <- list()
+  
+  var.names <- unlist(sapply(settings$state.data.assimilation$state.variable, 
+                             function(x) {
+                               x$variable.name
+                             }, 
+                             USE.NAMES = FALSE), 
+                      use.names = FALSE)
+  
+  start_date <- settings$state.data.assimilation$start.date
+  end_date   <- settings$state.data.assimilation$end.date
+  
+  obs.mean <- list()
+  obs.cov <- list()
+  
+  start.time <- format(lubridate::ymd(start_date),settings$state.data.assimilation$forecast.time.step)
+  end.time <- format(lubridate::ymd(end_date),settings$state.data.assimilation$forecast.time.step)
+  obs.times <- start.time:end.time
+  
+  
+  for(i in seq_along(format_id)){
+    input.list[[i]] <- db.query(paste("SELECT * FROM inputs WHERE site_id =",site$id ,"  AND format_id = ",format_id[[i]]), bety$con)
+    input.id[[i]] <- input.list[[i]]$id
+    
+    data.path <- PEcAn.DB::query.file.path(input.id[[i]], settings$host$name, bety$con)
+    format_full <- format <- PEcAn.DB::query.format.vars(input.id = input.id[[i]], bety, format.id = NA, var.ids=NA)
+    
+    format$na.strings <- 'NA'
+    
+    # ---- LOAD INPUT DATA ---- #
+    time.row <- format$time.row
+    vars.used.index <- setdiff(seq_along(format$vars$variable_id), format$time.row)
+    
+    print(paste('Using PEcAn.benchmark::load_data.R on format_id',format_id[[i]],'-- may take a few minutes'))
+    obvs[[i]] <- PEcAn.benchmark::load_data(data.path, format, start_year = lubridate::year(start_date), end_year = lubridate::year(end_date), site, vars.used.index, time.row)
+    
+    dataset <- obvs[[i]]
+    variable <- var.names
+    
+    ### Tree Ring Data Product
+    if(format_id[[i]] == '1000000040'){
+      obvs[[i]] <- obvs[[i]][obvs[[i]]$model_type=='Model RW + Census',]
+      obvs[[i]]$AbvGrndWood <- obvs[[i]]$AbvGrndWood * .48
+      obvs[[i]]$NPP_1_C <- obvs[[i]]$NPP_1_C * .48
+      arguments <- list(.(year, MCMC_iteration, site_id), .(variable))
+      arguments2 <- list(.(year), .(variable))
+      arguments3 <- list(.(MCMC_iteration), .(variable), .(year))
+    }else{
+      print('ERROR: This data format has not been added to this function (ツ)_/¯ ')
+      stop()
+    }
+    
+    ### Map species to model specific PFTs
+    if(any(var.names == 'AGB.pft')){
+      spp_id <- match_species_id(unique(dataset$species_id),format_name = 'usda',bety)
+      spp_id <- spp_id[spp_id$input_code!='HAVI4',]
```

Comment:
There are currently 6 PFTs that have witch hazel, do we don't want to delete it
```
     id     |                                         definition                                          |         created_at         |         updated_
at         |             name             | parent_id | pft_type | modeltype_id 
------------+---------------------------------------------------------------------------------------------+----------------------------+-----------------
-----------+------------------------------+-----------+----------+--------------
 2000000007 | Basic all plants PFT for the MAAT model.  Cloned from an ED2 example                        | 2016-04-11 15:00:39.008867 | 2016-04-11 15:03
:10.154362 | maat.all.plants              |        95 | plant    |   2000000001
         22 | Southern half of mid-successional temperate hardwoods (e.g. oak, hickory).  No default PFT. | 2011-02-01 20:41:20        | 2011-02-01 20:41
:20        | temperate.South_Mid_Hardwood |           | plant    |            1
         39 | all plants                                                                                  | 2011-09-27 15:53:10        | 2011-09-27 15:53
:10        | plants                       |           | plant    |            1
         83 | all plants                                                                                  | 2011-09-27 15:53:10        | 2015-01-06 14:03
:33.531563 | plants2                      |           | plant    |            1
         95 | all plants                                                                                  | 2011-09-27 15:53:10        | 2015-01-06 14:03
:53.548183 | plants3                      |           | plant    |            1
         96 | Growth respiration for all plants                                                           | 2011-09-27 15:53:10        | 2013-10-07 14:51
:52        | gr.plants                    |           | plant    |            1
(6 rows)
```
I think the more general question is what to do with ANY species that doesn't match the union of the species covered by the selected PFTs -- think about writing the code that way. Ideally, we don't want to loose biomass so it would be good to lump it into some PFT -- so the default should be to list the unmatched species and throw a logger.severe. But I could see having a flag that would allow you to drop unmatches species, especially when they're a tiny fraction of the community